go/store/prolly/tree/json_diff_test.go

```diff
@@ -0,0 +1,163 @@
+package tree
+
+import (
+	"io"
+	"testing"
+
+	"github.com/dolthub/go-mysql-server/sql/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+type jsonDiffTest struct {
+	name          string
+	from, to      types.JsonObject
+	expectedDiffs []JsonDiff
+}
+
+var simpleJsonDiffTests = []jsonDiffTest{
+	{
+		name:          "empty object, no modifications",
+		from:          types.JsonObject{},
+		to:            types.JsonObject{},
+		expectedDiffs: nil,
+	},
+	{
+		name: "insert into empty object",
+		from: types.JsonObject{},
+		to:   types.JsonObject{"a": 1},
+		expectedDiffs: []JsonDiff{
+			{
+				Key:  "$.\"a\"",
+				From: nil,
+				To:   &types.JSONDocument{Val: 1},
+				Type: AddedDiff,
+			},
+		},
+	},
+	{
+		name: "delete from object",
+		from: types.JsonObject{"a": 1},
+		to:   types.JsonObject{},
+		expectedDiffs: []JsonDiff{
+			{
+				Key:  "$.\"a\"",
+				From: &types.JSONDocument{Val: 1},
+				To:   nil,
+				Type: RemovedDiff,
+			},
+		},
+	},
+	{
+		name: "modify object",
+		from: types.JsonObject{"a": 1},
+		to:   types.JsonObject{"a": 2},
+		expectedDiffs: []JsonDiff{
+			{
+				Key:  "$.\"a\"",
+				From: &types.JSONDocument{Val: 1},
+				To:   &types.JSONDocument{Val: 2},
+				Type: ModifiedDiff,
+			},
+		},
+	},
+	{
+		name: "nested insert",
+		from: types.JsonObject{"a": types.JsonObject{}},
+		to:   types.JsonObject{"a": types.JsonObject{"b": 1}},
+		expectedDiffs: []JsonDiff{
+			{
+				Key:  "$.\"a\".\"b\"",
+				To:   &types.JSONDocument{Val: 1},
+				Type: AddedDiff,
+			},
+		},
```

Comment:
(minor) Just an idea, not critical... might be nice to have multiple expected diffs in a test case so that you could assert they get returned in the correct/expected order.